frontend/src/plugins/router.js

```diff
@@ -107,11 +108,21 @@ export default new Router({
                                             path: 'visitors',
                                             name: 'visitors',
                                             component: Visitors,
+                                            beforeEnter: (to, from, next) => {
```

Comment:
not quite sure that this check should happen here. We have route guard on layout itself to fetch data. this variant could cause case when this check will happen earlier than data will be fetched. Instead it's better to create mixin, and use meta to define what kind of access is needed for each particular page